checkpatch_hook/__init__.py

```diff
@@ -0,0 +1,252 @@
+            and not p.is_symlink()
+        ]
+        if patch_files:
+            commit_patch = _get_patch(running_in_ci, patch_files)
+            if commit_patch:
+                _run_checkpatch(commit_patch, errors, **post_dconfig)
+
+    # print errors
+    for filename, f_errors in errors.items():
+        print(f"{filename}:")
+        for error in f_errors:
+            print(f'  {error["line"]}: {error["message"]}')
+    if errors:
+        sys.exit(1)
+
+
+def _get_commit_files_from_log() -> list[Path]:
+    cmd = ["git", "diff-tree", "--no-commit-id", "--name-only", "HEAD", "-r"]
+    log_output = check_output(cmd).decode().rstrip("\n")
+    return [Path(file_path) for file_path in log_output.split("\n")]
+
+
+def _get_patch(running_in_ci: bool, patch_files: list[Path]) -> str:
+    if running_in_ci:
+        return check_output(
+            [
+                "git",
+                "format-patch",
+                "--stdout",
+                "-1",
+                "HEAD",
+                "--",
+                *patch_files,
+            ]
+        ).decode()
+    commit_msg = _read_commit_msg_from_file()
+    return _diff_to_patch(patch_files, commit_msg)
+
+
+def _diff_to_patch(patch_files: list[Path], commit_msg: str) -> str:
+    diff = check_output(["git", "diff", "--cached", "--", *patch_files]).decode()
+    # No changes, no patch
+    if not diff:
+        return ""
+    return PATCH_TEMPLATE.format(
+        diff=diff,
+        commit_msg=commit_msg,
+        current_datetime=datetime.now().strftime("%a, %d %b %Y %H:%M:%S %z"),
+    )
+
+
+def _is_relative_to(path: Path, dir_path: Path) -> bool:
+    try:
+        path.relative_to(dir_path)
+        return True
+    except ValueError:
+        return False
+
+
+def _read_commit_msg_from_file() -> str:
+    commit_msg_file = Path(".git") / "COMMIT_EDITMSG"
+    return commit_msg_file.read_text() if commit_msg_file.exists() else ""
+
+
+def _run_checkpatch(
+    commit_patch: str,
+    errors: DefaultDict[str, list],
+    errors_enabled: list | None = None,
+    errors_ignored: list | None = None,
+    max_line_length: int | None = None,
+) -> None:
+    cmd = [
+        "checkpatch.pl",
+        "--strict",  # Be more annoying
+        "--no-tree",  # No kernel source tree present
+        # Some stuff to make parsing easier (see below):
+        "--emacs",
+        "--terse",
+        "--showfile",
+        "--no-summary",
+        "--color=never",
+        "--no-signoff",  # Don't require Signed-Off-By
+    ]
+
+    if errors_ignored is not None:
+        cmd += ["--ignore", ",".join(errors_ignored)]
+    elif errors_enabled is not None:
+        cmd += ["--types", ",".join(errors_enabled)]
+    if max_line_length is not None:
+        cmd += [f"--max-line-length={max_line_length}"]
+
+    result = run(cmd, input=commit_patch.encode(), stdout=PIPE)
+
+    if result.returncode == 0:
+        return  # No errors
+    # There were errors, parse them
```

Comment:
Could this be tweaked a bit to move the parsing of the errors into a function?

So more like `if result.returncode != 0: # parse errors`?